### PR TITLE
KEYCLOAK-19711 Change Locations of h2-db files and tmp directory in Keycloak.X

### PR DESCRIPTION
**Pull request by DGuhr**

### CODE REVIEW

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/QuarkusPlatform.java [hunk 1]
```diff
@@ -35,6 +35,13 @@
 
     private static final Logger log = Logger.getLogger(QuarkusPlatform.class);
 
+    Runnable startupHook;
```
Comment (pedroigor) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/QuarkusPlatform.java:38]: I won't do code formatting until we have something in place that we can share. See https://issues.redhat.com/browse/KEYCLOAK-16322.

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/Environment.java [hunk 2]
```diff
@@ -60,6 +61,10 @@ public static Path getHomePath() {
         return null;
     }
 
+    public static String getDataDir() {
+        return System.getProperty("kc.home.dir") + DATA_PATH;
```
Comment (pedroigor) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/Environment.java:65]: Can we use `getHomeDir` instead?

Comment (DGuhr) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/Environment.java:65]: sure, will change

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/database/Database.java [hunk 3]
```diff
@@ -94,7 +94,7 @@ public static boolean isSupported(String alias) {
                 public String apply(String alias) {
                     if ("h2-file".equalsIgnoreCase(alias)) {
                         return "jdbc:h2:file:${kc.home.dir:${kc.db.url.path:~}}" + File.separator + "${kc.data.dir:data}"
-                            + File.separator + "keycloakdb${kc.db.url.properties:;;AUTO_SERVER=TRUE}";
+                            + File.separator + "h2/keycloakdb${kc.db.url.properties:;;AUTO_SERVER=TRUE}";
```
Comment (DGuhr) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/database/Database.java:97]: @pedroigor as i re-checked I'll also change this line to use File.seperator ;)

Comment (pedroigor) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/database/Database.java]: oops, missed that one :) Thanks.

**Review by pedroigor — APPROVED** [review]

**Pull request merged**